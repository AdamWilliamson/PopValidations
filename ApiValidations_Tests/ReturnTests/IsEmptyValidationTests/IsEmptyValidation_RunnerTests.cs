using ApprovalTests;
using FluentAssertions;
using ApiValidations_Tests.TestHelpers;
using ApiValidations;

namespace ApiValidations_Tests.ReturnTests.IsEmptyValidationTests.IsEmptyValidationTests;

public class EmptyApi
{
    public string ReturnMustBeEmpty() { return string.Empty; }
}

public class IsEmpty_TestingValidator : ApiValidator<EmptyApi>
{
    public IsEmpty_TestingValidator()
    {
        DescribeFunc(x => x.ReturnMustBeEmpty()).Return.IsEmpty();
    }
}

public class IsEmptyValidation_RunnerTests
{
    [Fact]
    public void WhenDescribing_ItReturnsTheValidation()
    {
        // Arrange
        var runner = ValidationRunnerHelper.BasicRunnerSetup(new IsEmpty_TestingValidator());

        // Act
        var description = runner.Describe();

        // Assert
        description.Results.Should().HaveCount(1);
        description.Results.Should().HaveCount(ValidatableHelper.GetValidatableCount<EmptyApi>(ValidatableType.NoExceptions));
        Approvals.VerifyJson(JsonConverter.ToJson(description));
    }
}
