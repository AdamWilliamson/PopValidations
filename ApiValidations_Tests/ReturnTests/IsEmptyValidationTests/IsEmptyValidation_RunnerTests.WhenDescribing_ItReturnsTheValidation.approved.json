{
    "Results": [
        {
            "Property": "ReturnMustBeEmpty()->String:Return(String)",
            "Outcomes": [
                {
                    "Validator": "IsEmptyValidation",
                    "Message": "Must be empty.",
                    "Values": []
                }
            ],
            "ValidationGroups": []
        }
    ]
}